### ISSUE TITLE
addStatements sometimes adds to the bottom of the file

### ISSUE THREAD

**Issue by chrisgervang** [dsherret/ts-morph#189]
I'm creating a new source file with addStatements, addImport, addVariableStatement, and addFunctions. All of them add in sequential order, except addStatements, which always seems to add to the last line of the file. I wanted to put this generated file warning comment at the top.

```
const mock = ast.createSourceFile(fileName)
mock.addStatements("/* --------\n  This is a generated file. All edits will be overwritten! \n-------- */")
mock.addVariableStatement({
          ...
})
mock.addImport({
         ...
})
mock.addFunctions(types.map(type => {
          ...
}))
mock.formatText()
mock.saveSync()
```

**Comment by dsherret** [comment]
It's not that addStatements always adds to the bottom, but that comments aren't tracked by the typescript compiler.

Temporary workaround is to do...

```typescript
mock.insertStatements(0, "/* --------\n  This is a generated file. All edits will be overwritten! \n-------- */");
```

...when you're all done.

What I could do to solve this is for any "add" method to insert after any comments.

**Comment by dsherret** [comment]
@rhys-vdw Sorry that message is not so helpful. It's hard to make that scenario work because JS docs are actually children of the node they are above and that method is expecting to insert a statement (that is, a child of the source file in this case).

To fix that issue, change the code to use the JS doc insert/add methods on that node: https://dsherret.github.io/ts-simple-ast/details/documentation

For example:

```ts
classDeclaration.addJsDoc(`Generated type guards for "${outFile.getRelativePathTo(sourceFile)}".`);
```

**Comment by rhys-vdw** [comment]
Ah right, I was trying to add a [module JSdoc](http://usejsdoc.org/tags-module.html), but perhaps that's not supported by TypeScript?

**Comment by dsherret** [comment]
Seems to not be. I was just trying it out here: https://ts-ast-viewer.com/

A way to work around this would be to do this:

```ts
// use a writer in order to easily add the correct newline based on the manipulation settings
sourceFile.insertText(0, writer =>
    writer.writeLine("/** Generated type guards for "${outFile.getRelativePathTo(sourceFile)}. */");
```

But just make sure to do that last because it will forget any previously navigated descendant nodes. Read more [here](https://dsherret.github.io/ts-simple-ast/manipulation/).

**Comment by dsherret** [comment]
Will be fixed in v2.0 due to #575.
